src/trackers/helpers/sharedData.js

```diff
@@ -18,28 +17,37 @@ class SharedData {
         this.domains = _getJSON(`${build}/generated/domain_summary.json`) || {}
         this.abuseScores = _getJSON(`${build}/generated/api_fingerprint_weights.json`)
         this.categories = _getCategories()
-        if (fs.existsSync(`${cfg.trackerDataLoc}/entities`)) {
-            this.domainToEntity = _readEntities()
-            this.entityMap = entityHelper.entityMap(`${cfg.trackerDataLoc}/entities`)
-        } else {
-            this.domainToEntity = {}
-            this.entityMap = new Map()
-        }
+        
+        const {domainToEntity, entityMap} = _readEntities(`${cfg.trackerDataLoc}/entities`)
+        this.domainToEntity = domainToEntity
+        this.entityMap = entityMap
+
         this.breaking = _getBreaking(`${build}/static/breaking`)
         this.topExampleSitesSet = _getTopExampleSites(cfg)
         this.nameservers = NameServers
     }
 }
 
 // map entity domains to name for easy lookup
-function _readEntities () {
-    return fs.readdirSync(`${config.trackerDataLoc}/entities/`).reduce((domainList, entityFile) => {
-        const entityData = _getJSON(`${config.trackerDataLoc}/entities/${entityFile}`)
-        entityData.properties.forEach(url => {
-            domainList[url] = entityData
+function _readEntities (path) {
+    const domainToEntity = {}
+    const entityMap = new Map()
+
+    if (fs.existsSync(path)) {
+        fs.readdirSync(`${config.trackerDataLoc}/entities/`).forEach(entityFile => {
+            const entityData = _getJSON(`${config.trackerDataLoc}/entities/${entityFile}`)
```

Comment:
Does this have to be sync? We could use a `Proxy` to load async, or even lazily. This way we avoid waiting for several thousand read operations before the script even starts.

Reply:
I agree and I thought about it, but I optimized reading of this data 2x (because we are no longer reading it twice) and TBH the result is barely visible.

before:
<img width="1406" alt="before-entityMap" src="https://user-images.githubusercontent.com/985504/106912302-319f0500-6703-11eb-96be-f6b60c9b3dbe.png">

after:
<img width="1438" alt="after-entityMap" src="https://user-images.githubusercontent.com/985504/106912169-0fa58280-6703-11eb-9051-195af2b42cb5.png">

We went down from 1800ms to 1100ms of startup. It doesn't seem worth going after this one second that's left.